napari/_vispy/vispy_text_layer.py

```diff
@@ -0,0 +1,152 @@
+import numpy as np
+from copy import copy
+from vispy.scene.visuals import Line, Compound
+from vispy.scene.visuals import Text as TextNode
+from .markers import Markers
+from vispy.visuals.transforms import ChainTransform
+
+from ..layers import Text
+from .vispy_base_layer import VispyBaseLayer
+import numpy as np
+
+
+class VispyTextLayer(VispyBaseLayer):
+    _highlight_color = (0, 0.6, 1)
+    _highlight_width = 1.5
+
+    _OUTLINE_MARKERS_NODE_INDEX = 0
+    _TEXT_NODE_INDEX = 1
+    _HIGHLIGHT_TEXT_NODE_INDEX = 2
+
+    def __init__(self, layer):
+        # Create a compound visual with the following thress subvisuals:
+        # Lines: The lines of the interaction box used for highlights.
+        # Text: The text to be diplayed.
+        # Text: Highlighted text to show which text is selected.
+        node = Compound([Line(), TextNode(), TextNode()])
+
+        super().__init__(layer, node)
+
+        self.layer.events.rotation.connect(lambda e: self._on_data_change())
+        self.layer.events.font_size.connect(lambda e: self._on_data_change())
+        self.layer.events.text_color.connect(lambda e: self._on_data_change())
+        self.layer.events.highlight.connect(
+            lambda e: self._on_highlight_change()
+        )
+
+        self.layer.dims.events.ndisplay.connect(
+            lambda e: self._on_display_change()
+        )
+
+        self._on_display_change()
+
+    def _on_display_change(self):
```

Comment:
i've improved how some of this is done recently to remove repeated calls to `_on_data_change`, see edits to Points #629, can you adapt those here too

Reply:
Got it. I'll work on this next.

Reply:
Done!